code/game/objects/structures/traps.dm

```diff
@@ -81,10 +85,79 @@
 	name = "shock trap"
 	desc = "A trap that will shock and render you immobile. You'd better avoid it."
 	icon_state = "trap-shock"
+	var/stun_time = 100
 
 /obj/structure/trap/stun/trap_effect(mob/living/L)
 	L.electrocute_act(30, src, safety=1) // electrocute act does a message.
-	L.Paralyze(100)
+	L.Paralyze(stun_time)
+
+/obj/structure/trap/stun/hunter
+	name = "bounty trap"
+	desc = "A trap that only goes off when a fugitive steps on it, announcing the location and stunning the target. You'd better avoid it."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "bounty_trap_on"
+	stun_time = 200
+	sparks = FALSE //the item version gives them off to prevent runtimes (see Destroy())
+	var/caught = FALSE
+
+/obj/structure/trap/stun/hunter/Initialize(mapload)
+	. = ..()
+	flare_message = "<span class='warning'>[src] snaps shut!</span>"
+
+/obj/structure/trap/stun/hunter/Destroy()
+	var/obj/item/bountytrap/item = new(get_turf(src))
+	item.name = name
+	if(caught)
+		item.snapshut()
+	. = ..()
+
+/obj/structure/trap/stun/hunter/Crossed(atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(!L.mind?.has_antag_datum(/datum/antagonist/fugitive))
+			return
+	caught = TRUE
+	. = ..()
+
+/obj/structure/trap/stun/hunter/flare()
+	..()
+	qdel(src)
+
+/obj/item/bountytrap
+	name = "bounty trap"
+	desc = "A trap that only goes off when a fugitive steps on it, announcing the location and stunning the target. It's currently inactive."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "bounty_trap_off"
+	var/obj/item/radio/radio
+	var/datum/effect_system/spark_spread/spark_system
+
+/obj/item/bountytrap/Initialize(mapload)
+	. = ..()
+	radio = new(src)
+	radio.subspace_transmission = TRUE
+	radio.canhear_range = 0
+	radio.recalculateChannels()
+	spark_system = new
+	spark_system.set_up(4,1,src)
+	spark_system.attach(src)
+	if(name == initial(name))
+		name = "[name] #[rand(1, 999)]"
+
+/obj/item/bountytrap/proc/snapshut()
+	spark_system.start()
+	playsound(src, 'sound/machines/ding.ogg', 50, 1)
+	radio.talk_into(src, "Fugitive has triggered this trap in the [get_area_name(src)]!", RADIO_CHANNEL_COMMON)
+
+/obj/item/bountytrap/attack_self(mob/living/user)
+	to_chat(user, "<span class=notice>You set up [src]. Examine while close to disarm it.</span>")
+	var/obj/structure/trap/stun/hunter/trap = new(get_turf(src))
+	trap.name = name
+	qdel(src)
```

Comment:
Why not stuff this in the trap, instead of remaking it every time ?

Reply:
there are quirks with cooldown, but i can do that.